File: media/textsamples/sample.go
// sample.go — Go syntax-highlighting sample for the UltraCanvas demo.
package main

import (
	"fmt"
	"math"
	"sort"
)

type Shape interface {
	Area() float64
	Name() string
}

type Circle struct {
	Radius float64
}

func (c Circle) Area() float64 { return math.Pi * c.Radius * c.Radius }
func (c Circle) Name() string  { return "circle" }

func main() {
	shapes := []Shape{
		Circle{Radius: 2.5},
		Circle{Radius: 1.0},
	}

	sort.Slice(shapes, func(i, j int) bool {
		return shapes[i].Area() < shapes[j].Area()
	})

	total := 0.0
	for _, s := range shapes {
		fmt.Printf("%s -> %.4f\n", s.Name(), s.Area())
		total += s.Area()
	}
	fmt.Printf("total = %.4f\n", total)
}

File: media/textsamples/sample.php
<?php
// sample.php — PHP syntax-highlighting sample for the UltraCanvas demo.
declare(strict_types=1);

interface Shape
{
    public function area(): float;
    public function name(): string;
}

final class Circle implements Shape
{
    public function __construct(private float $radius) {}

    public function area(): float
    {
        return M_PI * $this->radius ** 2;
    }

    public function name(): string
    {
        return 'circle';
    }
}

$shapes = [new Circle(1.0), new Circle(2.5)];

$total = array_reduce(
    $shapes,
    static fn (float $carry, Shape $s): float => $carry + $s->area(),
    0.0
);

foreach ($shapes as $shape) {
    printf("%s -> %.4f\n", $shape->name(), $shape->area());
}

printf("total = %.4f\n", $total);
